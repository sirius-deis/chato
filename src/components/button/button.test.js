import { render, screen, fireEvent } from '@testing-library/react';
import Button from './button';

describe('Button component', () => {
  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(
      <Button onClick={fn} kind='rounded' styles={{ fontSize: '2rem' }}>
        Click me
      </Button>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should render children with plain text', () => {
    render(<Button kind='rounded'>Click me</Button>);
    expect(screen.getByText('Click me')).toBeInTheDocument();
  });
  it('should render children with html element', () => {
    render(
      <Button kind='rounded'>
        <h2>Click me</h2>
      </Button>,
    );
    expect(screen.getByRole('button')).toContainHTML('<h2>Click me</h2>');
  });
  it('should fire an event', () => {
    const fn = jest.fn();
    render(
      <Button onClick={fn} kind='rounded'>
        Click me
      </Button>,
    );
    fireEvent.click(screen.getByRole('button'));
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledTimes(1);
  });
});
