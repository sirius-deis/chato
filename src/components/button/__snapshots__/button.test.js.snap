// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<div>
  <button
    class="sc-bgqQcB sc-ewnqHT coimH gsIsIn"
    style="font-size: 2rem;"
  >
    Click me
  </button>
</div>
`;
